server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1114,6 +1141,55 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
         return searchContext;
     }
 
+    private void writeTaskResourceUsage(SearchShardTask task) {
+        try {
+            // Get resource usages when task starts
```

Comment:
These were the previously-recorded resource usages from the start of the task, whereas we call this at the end of the task, right? 

If so, the comment should say something like "Get resource usages from when the task started".